api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentRecurringPatternServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+        String serverTimeZone = Calendar.getInstance().getTimeZone().getID();
+        TimeZone.setDefault(TimeZone.getTimeZone(clientTimeZone));
+        List<Appointment> pendingAppointments = getPendingOccurrences(appointment.getUuid(),
+                Arrays.asList(AppointmentStatus.Scheduled, AppointmentStatus.CheckedIn));
+        TimeZone.setDefault(TimeZone.getTimeZone(serverTimeZone));
+        pendingAppointments.stream()
+                .map(pendingAppointment -> {
+                    pendingAppointment.setStatus(appointmentStatus);
+                    setAppointmentAuditForStatusChange(pendingAppointment, onDate);
+                    appointmentDao.save(pendingAppointment);
+                    return pendingAppointment;
+                })
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public AppointmentRecurringPattern update(AppointmentRecurringPattern appointmentRecurringPattern) {
+        appointmentRecurringPatternDao.save(appointmentRecurringPattern);
+        return appointmentRecurringPattern;
+    }
+
+    private List<Appointment> getPendingOccurrences(String appointmentUuid, List<AppointmentStatus> applicableStatusList) {
+        Date startOfDay = getStartOfDay();
+        Appointment appointment = appointmentDao.getAppointmentByUuid(appointmentUuid);
+        return appointment.getAppointmentRecurringPattern().getAppointments()
+                .stream()
+                .filter(appointmentInList -> (appointmentInList.getStartDateTime().after(startOfDay)
+                        || startOfDay.equals(appointmentInList.getStartDateTime()))
+                        && applicableStatusList.contains(appointmentInList.getStatus()))
+                .collect(Collectors.toList());
+    }
+
+    private Date getStartOfDay() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.HOUR_OF_DAY, calendar.getMinimum(Calendar.HOUR_OF_DAY));
+        calendar.set(Calendar.MINUTE, calendar.getMinimum(Calendar.MINUTE));
+        calendar.set(Calendar.SECOND, calendar.getMinimum(Calendar.SECOND));
+        calendar.set(Calendar.MILLISECOND, calendar.getMinimum(Calendar.MILLISECOND));
+        return calendar.getTime();
+    }
+
+    private void setAppointmentAudit(Appointment appointment) {
+        try {
+            String notes = appointmentServiceHelper.getAppointmentAsJsonString(appointment);
+            updateAppointmentAudits(appointment, notes);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void setAppointmentAuditForStatusChange(Appointment appointment, Date onDate) {
+        String notes = onDate != null ? onDate.toInstant().toString() : null;
+        updateAppointmentAudits(appointment, notes);
+    }
+
+    private void updateAppointmentAudits(Appointment appointment, String notes) {
```

Comment:
this "updateAppointmentAudits" method has implementations in AppointmentRecurringPatternServiceImpl, AppointmentRecurringPatternServiceImpl and SingleAppointmentRecurringPatternMapper

Reply:
We have to move some part of code from api to omod. We can play a separate card for that where we can refactor this.